DPPP/BDABuffer.h

```diff
@@ -36,6 +38,9 @@ namespace DP3 {
     class BDABuffer
     {
     public:
+      enum class Field { kData, kFlags, kWeights, kFullResFlags, kBitCount };
+      using Fields = EnumBitset<Field>;
```

Comment:
The use of a EnumBitset to simulate a struct of boolean is too complicated. Why not just use struct Fields { bool hasData, hasFlags, hasWeights } ... etc ? (Or even inline these booleans in the main class). This adds lots of lines of code that people have to figure out to understand the semantics of a EnumBitset...

Reply:
- The semantics of EnumBitset are nearly equal to those of std::bitset: Instead of figuring out the semantics of a the Fields struct, you can use your existing knowledge about std::bitset.
- A separate struct Fields with 4 booleans easily gets equally complicated as the EnumBitset: You'll want a constructor for default initialization, and a Set() function for setting all bits. Only the operator[] members are not needed.
- An EnumBitset can be reused, since it's a template. Using separate structs will cause duplicate code.